addons/bus/static/src/services/presence_service.js

```diff
@@ -0,0 +1,70 @@
+/** @odoo-module **/
+
+import { browser } from '@web/core/browser/browser';
+import { registry } from '@web/core/registry';
+import core from 'web.core';
+
+export const presenceService = {
+    start(env) {
+        const LOCAL_STORAGE_PREFIX = 'presence';
+
+        // map window_focus event from the wowlBus to the legacy one.
+        env.bus.on('window_focus', null, isOdooFocused => {
+            core.bus.trigger('window_focus', isOdooFocused);
+        });
+
+        let isOdooFocused = true;
+        let lastPresenceTime = new Date().getTime();
+
+        function onFocusChange(isFocused) {
+            isOdooFocused = isFocused;
+            browser.localStorage.setItem(
+                `${LOCAL_STORAGE_PREFIX}.focus`,
+                isOdooFocused
+            );
+            if (isFocused) {
+                lastPresenceTime = new Date().getTime();
+                env.bus.trigger('window_focus', isOdooFocused);
+            }
+        }
+
+        function onPresence() {
+            lastPresenceTime = new Date().getTime();
+            browser.localStorage.setItem(
+                `${LOCAL_STORAGE_PREFIX}.lastPresence`,
+                lastPresenceTime
+            );
+        }
+
+        function onStorage({ key, newValue }) {
+            if (key === `${LOCAL_STORAGE_PREFIX}.focus`) {
+                isOdooFocused = newValue;
+                this.env.bus.trigger('window_focus', newValue);
+            }
+        }
+
+        browser.addEventListener('storage', onStorage);
+        browser.addEventListener('focus', onFocusChange.bind(true));
+        browser.addEventListener('blur', onFocusChange.bind(false));
+        browser.addEventListener('unload', onFocusChange.bind(false));
+        browser.addEventListener('click', onPresence);
+        browser.addEventListener('keydown', onPresence);
+
+        return {
+            getLastPresence() {
+                const lastSavedPresence = browser.localStorage.getItem(
+                    `${LOCAL_STORAGE_PREFIX}.lastPresence`
+                );
```

Comment:
Also if I remember well it's better to never try to actually read the value from the local storage if avoidable to prevent sync issues with some browsers.

Ideally value update from one tab to the other should happen only through the onStorage params.
And maybe just once at the start read the current value from the storage?